docs/contributor/adr/0000-istio-operator-design.md

```diff
@@ -0,0 +1,50 @@
+# Istio Operator Design
+
+Date: 2022-12-14
```

Comment:
We have the date only in this ADR and since we have it versioned it is not relevant. I'D remove it.